### PR TITLE
APPSRE-7516 saas has_test_jobs flag

### PR DESCRIPTION
**Pull request by fishi0x01**
Adding `has_test_jobs` to identify whether a saas file holds tests. Previously this was identified by `publish_job_logs`. However, that meaning is not very descriptive. `has_test_jobs` is a dedicated flag with clear meaning.

Note: we have a script that will add `hasTestJobs` flag to all saas files with `publishJobLogs` set.

Schema: https://github.com/app-sre/qontract-schemas/pull/700

### CODE REVIEW

File: reconcile/saas_auto_promotions_manager/utils/saas_files_inventory.py [hunk 1]
```diff
@@ -122,9 +122,7 @@ def _assemble_subscribers_with_auto_promotions(self) -> None:
                         blocked_versions=blocked_versions.get(
                             resource_template.url, set()
                         ),
-                        # Note: this will be refactored at a later point.
-                        # https://issues.redhat.com/browse/APPSRE-7516
-                        use_target_config_hash=bool(saas_file.publish_job_logs),
+                        use_target_config_hash=bool(saas_file.has_test_jobs),
```
Comment (hemslo) [inline reconcile/saas_auto_promotions_manager/utils/saas_files_inventory.py:125]: Is this the only usage of `has_test_jobs`? I don't know what features are different when `has_test_jobs` set to `True` or `False`.

Comment (fishi0x01) [reply reconcile/saas_auto_promotions_manager/utils/saas_files_inventory.py:125]: Currently usage of target_config_hash is the only difference

Comment (hemslo) [reply reconcile/saas_auto_promotions_manager/utils/saas_files_inventory.py:125]: When to set it to `True` or `False`? As a user, what can I get by setting it?

Comment (fishi0x01) [reply reconcile/saas_auto_promotions_manager/utils/saas_files_inventory.py:125]: If you set it you will have target_config_hash set in promotion_data

Comment (fishi0x01) [reply reconcile/saas_auto_promotions_manager/utils/saas_files_inventory.py:125]: This is existing functionality - nothing changes about that. This is just a more declarative flag, to remove hidden meaning of publishJobLogs

Comment (fishi0x01) [reply reconcile/saas_auto_promotions_manager/utils/saas_files_inventory.py:125]: More details are in the ticket https://issues.redhat.com/browse/APPSRE-7516

Comment (hemslo) [reply reconcile/saas_auto_promotions_manager/utils/saas_files_inventory.py:125]: Treat me like a first time user, I don't know anything about `publishJobLogs` or `has_test_jobs`, I guess by setting `publishJobLogs`, I can get the job logs somewhere, but I can't image what can I get by setting `has_test_jobs`. Do we really need it?

Comment (fishi0x01) [reply reconcile/saas_auto_promotions_manager/utils/saas_files_inventory.py:125]: This is described in detail in the ticket already

Comment (fishi0x01) [reply reconcile/saas_auto_promotions_manager/utils/saas_files_inventory.py:125]: If you have an alternate suggestion, please tell me

**Review by rporres — APPROVED** [review]
thanks for making this change. The hidden features of publish_job_logs were not great.

**Comment by fishi0x01** [comment]
After a chat with @hemslo we decided to refine this a little (more explicit flag name + maybe enabling this on per-target level). I will open a new schema PR shortly

**Pull request closed without merging**